daemon/start.go

```diff
@@ -45,7 +47,20 @@ func (daemon *Daemon) ContainerStart(name string, hostConfig *runconfig.HostConf
 	}
 
 	if err := container.Start(); err != nil {
-		return derr.ErrorCodeCantStart.WithArgs(name, utils.GetErrorMessage(err))
+		derrErr := derr.ErrorCodeCantStart.WithArgs(name, utils.GetErrorMessage(err))
```

Comment:
I don't really like this whole logic here, isn't it better to hide all of this inside `container.Start()`?